crypto/dilithium/pqcrystals_dilithium_ref_common/sign.c

```diff
@@ -196,6 +219,56 @@ int crypto_sign_signature(ml_dsa_params *params,
   return 0;
 }
 
+/*************************************************
+* Name:        crypto_sign_signature
+*
+* Description: FIPS 204: Algorithm 2 ML-DSA.Sign.
+*              Computes signature.
+*
+* Arguments:   - uint8_t *sig:   pointer to output signature (of length CRYPTO_BYTES)
+*              - size_t *siglen: pointer to output length of signature
+*              - uint8_t *m:     pointer to message to be signed
+*              - size_t mlen:    length of message
+*              - uint8_t *ctx:   pointer to contex string
+*              - size_t ctxlen:  length of contex string
+*              - uint8_t *sk:    pointer to bit-packed secret key
+*
+* Returns 0 (success) or -1 (context string too long)
+**************************************************/
+int crypto_sign_signature(ml_dsa_params *params,
+                          uint8_t *sig,
+                          size_t *siglen,
+                          const uint8_t *m,
+                          size_t mlen,
+                          const uint8_t *ctx,
+                          size_t ctxlen,
+                          const uint8_t *sk)
+{
+  size_t i;
+  uint8_t pre[257];
+  uint8_t rnd[RNDBYTES];
+
+  if(ctxlen > 255)
+    return -1;
+
+  /* Prepare pre = (0, ctxlen, ctx) */
+  pre[0] = 0;
+  pre[1] = ctxlen;
+  for(i = 0; i < ctxlen; i++)
+    pre[2 + i] = ctx[i];
```

Comment:
memcpy here too? Either way needs {}

Reply:
this was reference code, and I assume they did this to avoid a dependency on openssl. added in [54ca109](https://github.com/aws/aws-lc/pull/1999/commits/54ca1098fc3bc9a0629a24931e041dc72561bda6)